### PR TITLE
vrecord 2020-07-01

### PR DESCRIPTION
**Pull request by dericed**
Testing is not yet complete, but I'm posting this as a work-in-progress to show a public alternative to https://github.com/amiaopensource/homebrew-amiaos/pull/247. Please do not approve or merge until WIP status is removed.

### CODE REVIEW

File: vrecord.rb [hunk 1]
```diff
@@ -1,29 +1,32 @@
...
-  depends_on "gnuplot"
-  depends_on "mediaconch"
-  depends_on "mkvtoolnix"
-  depends_on "mpv"
-  depends_on "qcli"
-  depends_on "sdl"
-  depends_on "xmlstarlet"
+
+  on_macos do
+    depends_on "bash"
```
Comment (retokromer) [inline vrecord.rb:17]: Why is this necessary?

Comment (dericed) [reply vrecord.rb:17]: because the approach to using this [here document](https://github.com/amiaopensource/vrecord/blob/fb7ec511d37130556c11bc429d63a2ad13c4834e/vrecord#L268) doesn't work with the default bash of some versions of macos that we want to support.

Comment (retokromer) [reply vrecord.rb:17]: Thank you for the explanation! That’s true: the macOS 10.15 comes with a shitty version and I haven’t checked what is in 10.16 beta (according to the developper release) or 11.0 (according to the advertising). I didn’t got an error on this… because on my computers there is the last Bash.

I presume, you need at least 4? Should we not include a test of the minimal requested version?

**Comment by privatezero** [comment]
Hi @retokromer - regarding the `bash` dependency - I don't think we need a version test, as honestly it seems like macOS is the only context that seems to have these problems, and Bash will be a necessary install once Apple drops it, which is in the works, right?

What were the problems within Debian? I will test with Ubuntu tomorrow, but am curious on conflicts in a related flavor of Linux!

**Comment by retokromer** [comment]
@privatezero My idea was just to avoid installing the Homebrew Bash when an appropriate Bash is already installed on the computer… But it’s not blocking at all and I will mark this as resolved. (BTW: Homebrew recently switched to Bash for its installation on both Linux and macOS.)

**Comment by retokromer** [comment]
@privatezero Oops! Sorry, I forgot to answer your other point. I guess, I will have the time to delve into it again this afternoon. (It was not Bash ;-)

File: vrecord.rb [hunk 2]
```diff
@@ -1,29 +1,32 @@
...
-  sha256 "761aa248856eedc23619d41f5902bba8baba1c67bddc322fcaad7cb241a1edfc"
+  url "https://github.com/amiaopensource/vrecord/archive/v2020-06-26.tar.gz"
+  version "2020-06-26"
+  sha256 "e32b77f8da9ece0fe7ed2a0660fc67a1d6ef48ff8a4692161c57223cdaa3412d"
   head "https://github.com/amiaopensource/vrecord.git"
 
   bottle :unneeded
 
-  depends_on "amiaopensource/amiaos/decklinksdk"
+  # depends_on "amiaopensource/amiaos/deckcontrol"
```
Comment (privatezero) [inline vrecord.rb:11]: Conflict here seems fixed if merged upstream

File: vrecord.rb [hunk 3]
```diff
@@ -8,7 +8,7 @@ class Vrecord < Formula
 
   bottle :unneeded
```
Comment (retokromer) [inline vrecord.rb:9]: FYI: This is no longer useful. I will remove from all formulae in a PR next weekend.

File: vrecord.rb [hunk 4]
```diff
@@ -40,6 +40,20 @@ def install
     man1.install "vtest.1"
   end
 
+  on_linux do
+    def caveats
+    <<~EOS
+      ** IMPORTANT FOR LINUX INSTALL **
+      Additional install steps are necessary for a fully functioning Vrecord
+      install on Linux. This includes using the standard package manager to
+      install gnuplot, xmlstarlet, mkvtoolnix and mediaconch. Additionally,
```
Comment (retokromer) [inline vrecord.rb:49]: On my end, `gnuplot`, `xmlstarlet` and` mediaconch` are working as expected on Debian with Brew, only `mkvtoolnix` makes little troubles here (I presume, Moritz Bunkus knows that).

File: vrecord.rb (hunk 2 above)
Comment (dericed) [reply vrecord.rb:11]: ya, the new deckcontrol release should fix this part on linux

**Comment by dericed** [comment]
I forced an update. @privatezero I removed your `#` to deckcontrol. Please re-test.
